docs/spec/v2beta1/helmreleases.md

```diff
@@ -1270,6 +1270,42 @@ spec:
     crds: CreateReplace
 ```
 
+### Drift detection
+
+**Note:** This feature is experimental and can be enabled by setting `--feature-gates=DetectDrift=true`.
+
+When a HelmRelease is in-sync with the Helm release object in the storage, the controller will
+compare the manifests from the Helm storage with the current state of the cluster using a
+[server-side dry-run apply](https://kubernetes.io/docs/reference/using-api/server-side-apply/).
+If this comparison detects a drift (either due resource being created or modified during the
+dry-run), the controller will perform an upgrade for the release, restoring the desired state.
+
+### Excluding resources from drift detection
+
+The drift detection feature can be configured to exclude certain resources from the comparison
+by labeling or annotating them with `helm.toolkit.fluxcd.io/diff: disabled`. Using
```

Comment:
Open to suggestions to a better exclusion label, know that @stefanprodan wants to prevent "true" / "false" values and prefers something that matches with "enabled" / "disabled".

Reply:
Annotations are `camelCase`, but you're on the same mental journey as me.

Reply:
+1 on `helm.toolkit.fluxcd.io/drift-detection`.

Reply:
This would then become `helm.toolkit.fluxcd.io/driftDetection`, as we also have `reconcile.fluxcd.io/requestedAt`.